core/src/main/java/fr/sncf/osrd/envelope/Envelope.java

```diff
@@ -117,6 +120,18 @@ public double getMinSpeed() {
         return minSpeed;
     }
 
+    /** Returns the cumulative energy consumed at the wheels of the train, given a certain path and rolling stock */
+    public double getEnergyConsumed(
+            PhysicsPath path,
+            PhysicsRollingStock rollingStock
+    ) {
+        var cumulativeEnergy = 0.0;
+        for (var part : parts) {
+            cumulativeEnergy += part.getEnergyConsumed(path, rollingStock);
+        }
+        return cumulativeEnergy;
+    }
```

Comment:
This isn't the right place to add this helper: the `envelope` folder is a separate module from the `envelope_sim` module. Adding it as a static function somewhere else helps avoid a circular dependency. It does not matter now, but will when we will make the app more modular.

Reply:
You're right, that sounds like a better way to implement this. We'll change it.

Reply:
Moved to `EnvelopePhysics.java`.